notes.py

```diff
@@ -229,54 +234,80 @@ def parse_name(self, name: str) -> tuple[str, str, list[str], str]:
-            semitones_to_add = raise_octave.get(note_name, 0) * 12
+        for note_ind, note_name in enumerate(self.note_names):
+            semitones_to_add = raise_octave.get(note_ind, 0) * 12
             notes.append(lower.nearest_above(note_name).add_semitones(semitones_to_add))
             lower = notes[0]  # each subsequent note must be above root
         upper_chord = max(notes)  # extensions must be above chord
-        for note_name in self.extension_names:
-            semitones_to_add = raise_octave.get(note_name, 0) * 12
+        for note_ind_rel, note_name in enumerate(self.extension_names):
+            note_ind = note_ind_rel + len(self.note_names)
+            semitones_to_add = raise_octave.get(note_ind, 0) * 12
             notes.append(upper_chord.nearest_above(note_name).add_semitones(semitones_to_add))
         return Chord(notes)
 
-    def get_all_chords(self, *, lower: 'Note' = Note('C', 0), upper: 'Note') -> list['Chord']:
+    def get_all_chords(
+            self, *, lower: 'Note' = Note('C', 0), upper: 'Note',
+            allow_repeats: bool = False, max_notes: Optional[int] = None
+    ) -> list['Chord']:
         """
-        For a chord name, return all `Chord`s that can fit between `lower` and `upper`
+        For a chord name, return all `Chord`s that can fit between `lower` and `upper`;
+        If `allow_repeats`, chord notes (but not extensions) can be repeated
+        E.g., a G (G, B, D) could also be (G, B, D, G, D)
         """
 
         def _is_valid(notes: list[Note]) -> bool:
             return (
                 # All notes must fit between upper and lower
                 all(lower <= note <= upper for note in notes) and
                 # Root should be the lowest note (this should never be false based on `get_chord` definition)
-                all(notes[0] < other for other in notes[1:])
+                all(notes[0] <= other for other in notes[1:])
             )
-
         max_octaves = (upper - lower) // 12
-        # This is a list of dicts containing all the possible raise_octave combinations that might work
-        # There are actually a lot of invalid ones but those get handled by _is_valid
-        possible_raises = [
-            dict(zip(self.note_names + self.extension_names, combination))
-            for combination in product(range(max_octaves + 1), repeat=len(self.note_names) + len(self.extension_names))
-        ]
         chord_list = []
-        for raise_octave in possible_raises:
-            test_chord = self.get_chord(lower=lower, raise_octave=raise_octave)
-            if _is_valid(test_chord.notes):
-                chord_list.append(test_chord)
+        if allow_repeats:
+            note_sets = []
+            available_strings = max_notes - len(self.note_names + self.extension_names)
+            for repeat in range(available_strings + 1):
+                new_note_sets = [self.note_names + list(add) for add in combinations(self.note_names, r=repeat)]
```

Comment:
This is all the tuples of length between `len(self.note_names)` and `note_names` that contain at least `self.note_names` (but potentially also repeats)